src/Thumb.vue

```diff
@@ -0,0 +1,140 @@
+    <img ref="pic" @load.once="computeImgSize" :src="src" class="thumb-image" :width="computedWidth" :height="computedHeight" v-if="isImage">
+    <slot></slot>
+  </div>
+</template>
+
+<script>
+  const DEFAULT_SIZE = 100;
+
+  export default {
+    props: {
+      src: {
+        type: String
+      },
+      size: {
+        type: String,
+        default: ''
+      },
+      addClass: {
+        type: String,
+        default: ''
+      }, circle: {
+        type: Boolean,
+        default: false
+      },
+      border: {
+        type: String,
+        default: ''
+      },
+      text: {
+        type: String,
+        default: ''
+      },
+      fontsize: {
+        type: String,
+        default: ''
+      },
+      background: {
+        type: String,
+        default: ''
+      },
+      fontcolor: {
+        type: String,
+        default: ''
+      }
+    },
+    computed: {
+      isCircle () {
+        return this.circle !== undefined;
+      },
+      getBorder() {
+        // default is no border
+        return this.border === '' ? {} : {border: this.border};
+      },
+      getFontSize() {
+        // default is half of {size}
+        let fs = this.size/2;
+        fs = this.fontsize === '' ? fs : this.fontsize;
+        return {'font-size': fs + 'px'};
+      },
+      getFontColor() {
+        // default is no setting (inherit)
+        return this.fontcolor === '' ? {} : {'color': this.fontcolor};
+      },
+      getBgColor() {
+        // default is no setting (inherit)
+        return this.background === '' ? {} : {'background': this.background};
+      }
+    },
+    data () {
+      return {
+        computedWidth:'',
+        computedHeight:''
+      }
+    },
+    beforeCreate() {
+      console.log(this)
+    },
+    mounted() {
+      // TODO: Add text texts
+      // TODO: Independent scaling
+      this.size = this.size || 100;
+      let wrapperElement = this.$el;
+      console.log("thumb", this.size, this.size);
+      console.log("thumb circle", this.circle);
+      wrapperElement.style.width = this.size + 'px';
+      wrapperElement.style.height = this.size + 'px';
+      wrapperElement.style.lineHeight = this.size + 'px';
+      wrapperElement.style.fontsize = Math.round(this.size/2) + 'px';
+    },
+    methods: {
+      computeImgSize() {
+        // Scale the img within such that the cropped image is the required size
+        const renderedImg = this.$refs.pic;
+        const imgHeight = renderedImg.naturalHeight;
+        const imgWidth = renderedImg.naturalWidth;
+        const imgSize = Math.min(imgHeight, imgWidth);
+        const expansionFactor = this.size/imgSize;
+
+        this.computedWidth = imgWidth * expansionFactor;
+        this.computedHeight = imgHeight * expansionFactor;
+      },
+      isImage() {
+        // returns true if src is defined and is not an empty string
+        // TODO: Add cheerio/markdown preprocessing for emojis in markbind/page.js
```

Comment:
Is emoji support really required? Since there are alternative ways to enlarge an emoji and use it as a thumbnail within MarkBind. One way is to use MarkBind variables feature, wrap the emoji with a span and custom CSS to achieve the same result.

Reply:
The support for emoji, icons and text are all bundled together due to using the Markdown parser, and this features gives users a simpler, pre-built solution